### ISSUE TITLE
Component: Carousel - Navigators do not enable when paging through with a keyboard

### ISSUE THREAD

**Issue by coughenberg** [primefaces/primeng#13171]
### Describe the bug

When focusing on the carousel, you are able to press the Tab button in order to go to the next page for the carousel. But, if you are on the first page and tab to the next page, the Previous Item Arrow is still disabled and unable to be used unless you click on the Next Item Arrow to refresh the disable check on the Previous Item Arrow. 

Similar for if you are on the last page and press shift + tab to go back a page on the carousel, the Next Item Arrow does not get enabled.

Using keyboard support in the Carousel also does not update the Indicators if they are present on the Carousel.

This can be seen on all demos of the Carousel on the main site 
https://primeng.org/carousel

### Environment

Attempted on Chrome and Firefox

### Reproducer

https://primeng.org/carousel

### Angular version

16.0.0

### PrimeNG version

16.0.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

18.6.0

### Steps to reproduce the behavior

1. Go to https://primeng.org/carousel
2. Note the Indicator at the bottom is on page 1 and the Back Navigator is disabled
3. Click on the carousel to focus on it
4. Press "Tab" until the carousel goes to the next page
5. While on the next page, the Indicator has not updated and the Back Navigator is still disabled

### Expected behavior

The Indicator and Back Navigator should be updated while using keyboard controls to paginate through the PrimeNg Carousel.
